### PR TITLE
tests: add unit tests for sigstore verifier and signer logic

### PR DESCRIPTION
**Pull request by pushkarscripts**
## Description

Related to #1259

Adds unit tests to improve coverage for Sigstore verifier and signer logic in `internal/signerverifier/sigstore`.

The tests cover:
- Verifier behavior (`KeyID`, `ExpectedExtensionKind`, `SetExtension`)
- Verifier construction via `NewVerifierFromIdentityAndIssuer` (default, with options, and edge cases)
- Signer behavior, including option propagation and `KeyID`
- Metadata key generation (`MetadataKey`)
- Basic workflow-level coverage for `Sign` and `Verify`

The tests now attempt to move closer to the actual signing and verification workflow. Since Sigstore depends on external services (e.g., Rekor/Fulcio), these tests currently validate expected failure behavior in this setup rather than full end-to-end verification.

## AI Usage

- [ ] I **did not** use generative AI at all in making the content of this pull
  request.
- [x] I **did** use generative AI in some form in making the content of this
  pull request. I have described my use of AI below.

I used generative AI for general guidance while structuring tests and thinking about coverage. All code and tests were written, reviewed, and understood by me before submission.

## Contributor Checklist

- [x] I **have manually reviewed all content** submitted to gittuf in this pull
  request.
- [x] I fully understand the content I am submitting.
- [x] The changes introduced are documented and have tests included if
  applicable.
- [x] My changes do not infringe on copyright/trademarks/etc.
- [x] All commits in this pull request include a [DCO
  Signoff](https://wiki.linuxfoundation.org/dco).
- [x] By submitting this pull request, I agree to follow the gittuf [Code of
  Conduct](https://github.com/gittuf/community/blob/main/CODE-OF-CONDUCT.md).

### CODE REVIEW

**Review by patzielinski — COMMENTED** [review]
I'm not sure if this meaningfully tests our Sigstore workflow. The main concern I have is that we don't test any kind of signing or verification, which is done for other methods, e.g. GPG: https://github.com/gittuf/gittuf/blob/main/internal/signerverifier/gpg/gpg_test.go.

This is what I was referring to in the issue about using Sigstore's package to possibly mock the API or just spin up a Rekor server here.

**Comment by pushkarscripts** [comment]
I’ve updated the patch to move a bit closer to the signing and verification workflow by exercising `Sign` and `Verify`. Since Sigstore depends on external services, the current tests mostly cover the expected failure paths in this setup.

I’ll spend some time next looking into mocking Sigstore interactions or using a local Rekor instance so the full signing and verification flow can be tested more realistically.

**Comment by pushkarscripts** [comment]
@patzielinski @adityasaky just checking in on this PR. Would appreciate any feedback. Thanks!

File: internal/signerverifier/sigstore/options/signer/signer.go [hunk 1]
```diff
@@ -11,11 +17,15 @@ const (
...
-	ClientID    string
-	RedirectURL string
-	FulcioURL   string
-	RekorURL    string
+	IssuerURL     string
+	ClientID      string
+	RedirectURL   string
+	FulcioURL     string
+	RekorURL      string
+	IDTokenGetter oauthflow.TokenGetter
```
Comment (wlynch) [inline internal/signerverifier/sigstore/options/signer/signer.go]: I don't think we're setting this anywhere today - remove?

File: internal/signerverifier/sigstore/sigstore_test.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		signer.rekor = &mockTransparency{}
+
+		_, err = signer.Sign(context.Background(), data)
+		// It will fail because our mocks return errors, but it confirms we hit the sign path
+		// and that identity/issuer were correctly parsed from the token.
+		assert.Error(t, err)
+
+		assert.Equal(t, identity, signer.identity)
+		assert.Equal(t, issuer, signer.issuer)
+	})
```
Comment (wlynch) [inline internal/signerverifier/sigstore/sigstore_test.go]: This is mostly equivalent to 1 above (1 uses an in-memory implementation while this uses a mock). I don't think this gives use that much more, so we can omit this.

File: internal/signerverifier/sigstore/sigstore_test.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	virtualSigstore, err := ca.NewVirtualSigstore()
+	require.NoError(t, err)
+
+	identity := "user@example.com"
+	issuer := "https://issuer.example.com"
+	data := []byte("test data")
+
+	// 1. Test Verification with externally signed artifact
+	t.Run("Verify externally signed artifact", func(t *testing.T) {
+		entity, err := virtualSigstore.Sign(identity, issuer, data)
```
Comment (wlynch) [inline internal/signerverifier/sigstore/sigstore_test.go:135]: Instead of calling virtualSigstore.Sign, call gittuf's Signer.Sign. This should also fix the extension issue and let you actually be able to verify.

File: internal/signerverifier/sigstore/options/signer/signer.go [hunk 4]
```diff
@@ -11,11 +17,15 @@ const (
...
+	Fulcio        sign.CertificateProvider
```
Comment (wlynch) [inline internal/signerverifier/sigstore/options/signer/signer.go]: Add documentation that this takes priority over FulcioURL (same with rekor)

**Comment by pushkarscripts** [comment]
Thanks @wlynch for the review!

I've addressed the comments about the unused ID token injection support and added documentation clarifying that the injected Fulcio/Rekor implementations take precedence over the configured URLs. I also removed the redundant signer mock test.

I'm still working through the remaining feedback about using gittuf's `Signer.Sign()` instead of `virtualSigstore.Sign()` for the workflow test and will follow up with an update there.

File: internal/signerverifier/sigstore/sigstore_test.go (hunk 3 above)
Comment (pushkarscripts) [reply internal/signerverifier/sigstore/sigstore_test.go:135]: > Instead of calling virtualSigstore.Sign, call gittuf's Signer.Sign. This should also fix the extension issue and let you actually be able to verify.

I've been looking into this but got a bit stuck.

`VirtualSigstore` implements `root.TrustedMaterial`, but I don't see a way to use it directly with `Signer.Sign()` since it doesn't implement `sign.CertificateProvider` or `sign.Transparency`.

Am I missing something, or should I add test-only wrappers for those interfaces?
